src/v0/destinations/criteo_audience/networkHandler.js

```diff
@@ -0,0 +1,74 @@
+const {
+  prepareProxyRequest,
+  proxyRequest
+} = require("../../../adapters/network");
+const { isHttpStatusSuccess } = require("../../util/index");
+const {
+  REFRESH_TOKEN
+} = require("../../../adapters/networkhandler/authConstants");
+const tags = require("../../util/tags");
+const {
+  getDynamicErrorType,
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const { NetworkError } = require("../../util/errorTypes");
+
+/**
+ * This function helps to determine type of error occured. According to the response
+ * we set authErrorCategory to take decision if we need to refresh the access_token
+ * or need to disable the destination.
+ * @param {*} code
+ * @param {*} response
+ * @returns
+ */
+const getAuthErrCategory = code => {
+  switch (code) {
+    // https://developers.criteo.com/marketing-solutions/docs/api-error-types#:~:text=If%20Criteo%20API%20responded%20with%20401%20it%20means
+    case 401:
```

Comment:
Don't we need to check the response along with the code? 401 can come for invalid credentials as well.

Reply:
incorporated the` error[0].code` along with `status`